pkg/devfile/types.go

```diff
@@ -0,0 +1,20 @@
+package devfile
+
+// Schema is the index file schema for Samples
+type Schema struct {
+	Name        string            `yaml:"name,omitempty" json:"name,omitempty"`
+	DisplayName string            `yaml:"displayName,omitempty" json:"displayName,omitempty"`
+	Description string            `yaml:"description,omitempty" json:"description,omitempty"`
+	Tags        []string          `yaml:"tags,omitempty" json:"tags,omitempty"`
+	Icon        string            `yaml:"icon,omitempty" json:"icon,omitempty"`
+	Type        string            `yaml:"type,omitempty" json:"type,omitempty"`
```

Comment:
Do we want to support any type? or should be have an enum here ?

Reply:
Good point, I'll pass this along to the https://github.com/devfile/registry  folks when they're writing their schema for the index entries but this info is not to be used in ODC. I have just put it here to perform some simple go test, Discussion about type brought up in this PR too - https://github.com/openshift/console/pull/8687#discussion_r620493235

Reply:
FYI the actual schema for the index is here, so I'll get them to switch to enum, good point 👍 
https://github.com/devfile/registry-support/blob/master/index/generator/schema/schema.go#L79

Reply:
Please do, thanks 👍

Reply:
@maysunfaisal it partially does, but we need to use this type here as well, that was my main concern. I would suggest to vendor the `schema.go` here.

Reply:
@jhadvig I've added the enum as type to this PR. I am not going to vendor the other repo yet as its still in development phase and they dont have all the work done yet.